### ISSUE TITLE
"undefined method `test_target_type?` error with "get_version_number"

### ISSUE THREAD

**Issue by faithfracture** [fastlane/fastlane#12215]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

<details>
  <pre>$ bundle exec fastlane myapp

[✔] 🚀
+--------------------------------------+---------+--------------------------------------+
|                                     Used plugins                                      |
+--------------------------------------+---------+--------------------------------------+
| Plugin                               | Version | Action                               |
+--------------------------------------+---------+--------------------------------------+
| fastlane-plugin-update_provisioning  | 1.3.0   | update_provisioning_profile_specifi  |
| _profile_specifier                   |         | er                                   |
| fastlane-plugin-badge                | 1.0.0   | add_badge                            |
+--------------------------------------+---------+--------------------------------------+

[13:26:11]: Driving the lane 'myapp' 🚀
[13:26:11]: -------------------
[13:26:11]: --- Step: is_ci ---
[13:26:11]: -------------------
[13:26:11]: --------------------------------
[13:26:11]: --- Step: get_version_number ---
[13:26:11]: --------------------------------
+---------------+---------+
|      Lane Context       |
+---------------+---------+
| PLATFORM_NAME |         |
| LANE_NAME     | myapp  |
+---------------+---------+
[13:26:11]: undefined method `test_target_type?' for #<Xcodeproj::Project::Object::PBXAggregateTarget:0x00007f9134202598>

+------+--------------------+-------------+
|            fastlane summary             |
+------+--------------------+-------------+
| Step | Action             | Time (in s) |
+------+--------------------+-------------+
| 1    | is_ci              | 0           |
| 💥   | get_version_number | 0           |
+------+--------------------+-------------+

[13:26:11]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

Found no similar issues. To create a new issue, please visit:
https://github.com/fastlane/fastlane/issues/new
Run `fastlane env` to append the fastlane environment to your issue
bundler: failed to load command: fastlane (/Users/someguy/.rbenv/versions/2.5.1/bin/fastlane)
NoMethodError: [!] undefined method `test_target_type?' for #<Xcodeproj::Project::Object::PBXAggregateTarget:0x00007f9134202598>
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane/lib/fastlane/actions/get_version_number.rb:41:in `reject'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane/lib/fastlane/actions/get_version_number.rb:41:in `get_target!'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane/lib/fastlane/actions/get_version_number.rb:13:in `run'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane/lib/fastlane/runner.rb:257:in `block (2 levels) in execute_action'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane/lib/fastlane/runner.rb:236:in `block in execute_action'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane/lib/fastlane/runner.rb:231:in `chdir'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane/lib/fastlane/runner.rb:231:in `execute_action'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane/lib/fastlane/fast_file.rb:148:in `method_missing'
  Fastfile:59:in `set_build_number'
  Fastfile:77:in `doBuild'
  Fastfile:177:in `block in parsing_binding'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane/lib/fastlane/lane.rb:33:in `call'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane/lib/fastlane/lane_manager.rb:57:in `cruise_lane'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:75:in `run!'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/fastlane/lib/fastlane/cli_tools_distributor.rb:107:in `take_off'
  /Users/someguy/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.88.0/bin/fastlane:20:in `<top (required)>'
  /Users/someguy/.rbenv/versions/2.5.1/bin/fastlane:23:in `load'
  /Users/someguy/.rbenv/versions/2.5.1/bin/fastlane:23:in `<top (required)>'
</pre>
</details>

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13.4                                     |
| Ruby                        | 2.5.1                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.16.3                          |
| Installation Source         | ~/.rbenv/versions/2.5.1/bin/fastlane        |
| Host                        | Mac OS X 10.13.4 (17E199)                   |
| Ruby Lib Dir                | ~/.rbenv/versions/2.5.1/lib                 |
| OpenSSL Version             | OpenSSL 1.0.2o  27 Mar 2018                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.3                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./.fastlane/Fastfile`</summary>

```ruby
IosProject  = "MyIosProject/MyIosProject.xcodeproj"

MyTeamId   = '#####'
MyItcId    = '#####'
MyAppId    = 'com.myapp.appid'
MyScheme   = "MyScheme"
MyPlist    = "SupportingFiles/MyApp-Info.plist"
MyTvScheme = "MyTvApp"
MyTvPlist  = "SupportingFiles/MyApp-TV-Info.plist"

def set_build_number(options, project)
  versionNum = get_version_number(xcodeproj: project)
  buildNum = options[:buildNumber] || 1
  increment_build_number(xcodeproj: project, build_number: buildNum)
  "#{versionNum}.#{buildNum}"
end

def switch_beta(project, scheme, plist)
  identifier = CredentialsManager::AppfileConfig.try_fetch_value(:app_identifier)
  update_app_identifier(app_identifier:identifier, plist_path:plist, xcodeproj:project)
  update_provisioning_profile_specifier(xcodeproj: project, target: scheme, new_specifier: ' Beta', append: true)
  add_badge
end

def doBuild(options, project, scheme)
  fullVersion = set_build_number(options, project)
  outputFile = scheme
  outputFile += "-pre" unless !options[:beta]
  outputFile += "-#{fullVersion}"

  sigh

  gym(
    scheme: scheme, 
    clean: true,
    output_name: outputFile,
  )
  pilot(skip_submission: true, skip_waiting_for_build_processing: true) unless !options[:deploy]
end

# MyApp iOS
desc "Build the MyApp Mobile iOS app"
lane :myapp do |options|
  ENV["TEAM_ID"] = MyTeamId
  ENV["ITC_ID"] = MyItcId
  ENV["APP_IDENTIFIER"] = MyAppId
  if options[:beta]
    ENV["APP_IDENTIFIER"] = MyAppId + '.beta'
    switch_beta(IosProject, MyScheme, MyPlist)
  end
  doBuild(options, IosProject, MyScheme) unless options[:configOnly]
end

# MyApp tvOS
desc "Build the MyApp tvOS app"
lane :myapp_tv do |options|
  ENV["TEAM_ID"] = MyTeamId
  ENV["ITC_ID"] = MyItcId
  ENV["APP_IDENTIFIER"] = MyAppId
  if options[:beta]
    ENV["APP_IDENTIFIER"] = MyAppId + '.beta'
    switch_beta(TvosProject, MyTvScheme, MyTvPlist)
  end
  doBuild(options, TvosProject, MyTvScheme) unless options[:configOnly]
end


```
</details>

<details><summary>`./.fastlane/Appfile`</summary>

```ruby
app_identifier ENV["APP_IDENTIFIER"]
team_id ENV["TEAM_ID"]
itc_team_id ENV["ITC_ID"]
apple_id 'myappleid'


```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.88.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                                                | Version | Update-Status |
| ----------------------------------------------------- | ------- | ------------- |
| fastlane-plugin-update_provisioning_profile_specifier | 1.3.0   | ✅ Up-To-Date  |
| fastlane-plugin-badge                                 | 1.0.0   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                                                   | Version      |
| ----------------------------------------------------- | ------------ |
| did_you_mean                                          | 1.2.0        |
| slack-notifier                                        | 2.3.2        |
| atomos                                                | 0.1.2        |
| CFPropertyList                                        | 3.0.0        |
| claide                                                | 1.0.2        |
| colored2                                              | 3.1.2        |
| nanaimo                                               | 0.2.4        |
| xcodeproj                                             | 1.5.7        |
| rouge                                                 | 2.0.7        |
| xcpretty                                              | 0.2.8        |
| terminal-notifier                                     | 1.8.0        |
| unicode-display_width                                 | 1.3.0        |
| terminal-table                                        | 1.8.0        |
| plist                                                 | 3.4.0        |
| public_suffix                                         | 2.0.5        |
| addressable                                           | 2.5.2        |
| multipart-post                                        | 2.0.0        |
| word_wrap                                             | 1.0.0        |
| tty-screen                                            | 0.6.4        |
| tty-cursor                                            | 0.5.0        |
| tty-spinner                                           | 0.8.0        |
| babosa                                                | 1.0.2        |
| colored                                               | 1.2          |
| highline                                              | 1.7.10       |
| commander-fastlane                                    | 4.4.6        |
| excon                                                 | 0.62.0       |
| faraday                                               | 0.14.0       |
| unf_ext                                               | 0.0.7.5      |
| unf                                                   | 0.1.4        |
| domain_name                                           | 0.5.20170404 |
| http-cookie                                           | 1.0.3        |
| faraday-cookie_jar                                    | 0.0.6        |
| fastimage                                             | 2.1.1        |
| gh_inspector                                          | 1.1.3        |
| json                                                  | 2.1.0        |
| mini_magick                                           | 4.5.1        |
| multi_json                                            | 1.13.1       |
| multi_xml                                             | 0.6.0        |
| rubyzip                                               | 1.2.1        |
| security                                              | 0.1.3        |
| xcpretty-travis-formatter                             | 1.0.0        |
| dotenv                                                | 2.2.1        |
| bundler                                               | 1.16.1       |
| faraday_middleware                                    | 0.12.2       |
| naturally                                             | 2.1.0        |
| simctl                                                | 1.6.3        |
| uber                                                  | 0.1.0        |
| declarative                                           | 0.0.10       |
| declarative-option                                    | 0.1.0        |
| representable                                         | 3.0.4        |
| retriable                                             | 3.1.1        |
| mime-types-data                                       | 3.2016.0521  |
| mime-types                                            | 3.1          |
| little-plugger                                        | 1.1.4        |
| logging                                               | 2.2.2        |
| jwt                                                   | 2.1.0        |
| memoist                                               | 0.16.0       |
| os                                                    | 0.9.6        |
| signet                                                | 0.8.1        |
| googleauth                                            | 0.6.2        |
| httpclient                                            | 2.8.3        |
| google-api-client                                     | 0.13.6       |
| openssl                                               | 2.1.0        |
| date                                                  | 1.0.0        |
| fileutils                                             | 1.0.2        |
| ipaddr                                                | 1.2.0        |
| fastlane-plugin-update_provisioning_profile_specifier | 1.3.0        |
| curb                                                  | 0.9.4        |
| badge                                                 | 0.8.7        |
| fastlane-plugin-badge                                 | 1.0.0        |
</details>

*generated on:* **2018-04-02**
</details>

**Comment by AtomicCactus** [comment]
Same, happening after updating to the latest version

**Comment by joshdholtz** [comment]
Fixed in #12212. Making a release tonight 😊
